server/src/main/java/org/elasticsearch/search/lookup/SourceLookup.java

```diff
@@ -32,24 +32,10 @@
 
 import static java.util.Collections.emptyMap;
 
-public class SourceLookup implements Map<String, Object> {
+public interface SourceLookup extends Map<String, Object> {
```

Comment:
I *wonder* if we should/can confine the `Map` extension to scripts. It'd an odd thing to do outside of script compatibility and this is used so many places. It's probably not worth it for this PR, but it's an interesting question.